cli/azd/pkg/templates/json_source.go

```diff
@@ -5,13 +5,13 @@ import (
 	"fmt"
 )
 
-// NewJsonTemplateSource creates a new template source from a JSON string.
-func NewJsonTemplateSource(name string, jsonTemplates string) (Source, error) {
+// newJsonTemplateSource creates a new template source from a JSON string.
+func newJsonTemplateSource(name string, jsonTemplates string) (Source, error) {
```

Comment:
What is the harm of exporting these constructor functions? I'd prefer we just leave them as they were.

Reply:
it's about self-documenting the code.  
When we export, we are exposing 2 ways to using the package, one from the top abstraction manager and another from the functions directly.
Since we are using a main entry point for the package by using a Manager that handles the template source creation as implementation detail, the sources should not be exported.

It's just about enforcing the usage we want for the package, so we know there's less changes of bugs in the future is someone bridges the Manager and uses the template source directly.

If we want to export, then we might not need a Manager on top of the template sources